lua/ttt2/libraries/events.lua

```diff
@@ -0,0 +1,387 @@
+-- that satisfy the data from the first two layers.
+-- @return table A table with the reordered events
+-- @realm shared
+function events.SortByPlayerAndEvent()
+	local eventList = events.list
+	local sortedList = {}
+
+	for i = 1, #eventList do
+		local event = eventList[i]
+		local type = event.type
+		local plys64 = event:GetAffectedPlayer()
+
+		-- now that we have a list of all players affected by this event
+		-- each of those players should have this event added to their list
+		for k = 1, #plys64 do
+			local ply64 = plys64[k]
+
+			sortedList[ply64] = sortedList[ply64] or {}
+			sortedList[ply64][type] = sortedList[ply64][type] or {}
+
+			sortedList[ply64][type][#sortedList[ply64][type] + 1] = event
+		end
+	end
+
+	return sortedList
+end
+
+---
+-- Returns a table with the player steamID64 as indexes and the score
+-- for this specific player.
+-- @note Players with zero score this round will not be included in this list.
+-- @return table A table with the score per player
+-- @realm shared
+function events.GetTotalPlayerScores()
+	local eventList = events.list
+	local scoreList = {}
+
+	for i = 1, #eventList do
+		local event = eventList[i]
+
+		if not event:HasScore() then continue end
+
+		local plys64 = event:GetScoredPlayers()
+
+		for k = 1, #plys64 do
+			local ply64 = plys64[k]
+
+			scoreList[ply64] = (scoreList[ply64] or 0) + event:GetSummedPlayerScore(ply64)
+		end
+	end
+
+	return scoreList
+end
+
+---
+-- Returns a table with the player steamID64 as indexes and a number of deaths
+-- for this specific player.
+-- @note Players with zero deaths this round will not be included in this list.
+-- @return table A table with the amounts of deaths per player
+-- @realm shared
+function events.GetTotalPlayerDeaths()
+	local eventList = events.list
+	local deathList = {}
+
+	for i = 1, #eventList do
+		local event = eventList[i]
+
+		if event.type ~= EVENT_KILL then continue end
+
+		local victim64 = event.event.victim.sid64
+
+		deathList[victim64] = (deathList[victim64] or 0) + 1
+	end
+
+	return deathList
+end
+
+---
+-- Generates an event list in the deprecated format. Only contains
+-- events that were present in default TTT. Is sorted by time.
+-- @return table The deprecated event list
+-- @realm shared
+function events.GetDeprecatedEventList()
+	local deprecatedEvents = {}
+
+	for i = 1, #events.list do
+		local event = events.list[i]
+		local deprecatedEventData = event:GetDeprecatedFormat()
+
+		if not deprecatedEventData then continue end
+
+		deprecatedEvents[#deprecatedEvents + 1] = deprecatedEventData
+	end
+
+	return deprecatedEvents
+end
+
+if SERVER then
+	---
+	-- Triggers an event, adds it to a managed list and starts the score calculation for this event.
+	-- @param string name The name identifer of the event
+	-- @param any ... The arguments that should be passed to the event, see the @{EVENT:Trigger} function
+	-- @realm server
+	function events.Trigger(name, ...)
```

Comment:
I really dislike using varargs and we should discuss, if it is not more reasonable to use a table instead

Reply:
Hm, I think varargs are fine here for two reasons:
- the usage is identical to hooks
- events aren't called in draw hooks etc

Reply:
Then its fine